store/sqlstore/post_store.go

```diff
@@ -1780,3 +1790,34 @@ func (s *SqlPostStore) SearchPostsInTeamForUser(paramsList []*model.SearchParams
 
 	return model.MakePostSearchResults(posts, nil), nil
 }
+
+func wrapEmails(terms string) (string, bool) {
+	emailRegex := regexp.MustCompile("^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")
+	var termsWithWrappedEmails []string
+	containsEmails := false
+	// remove quotes if it has
+	termsWithoutQuotes := terms
+	if len(termsWithoutQuotes) > 0 && termsWithoutQuotes[0] == '"' {
```

Comment:
Here are you checking for quotes for the whole string and for that you can use `strings.Trim`. With the simpler regexp, you can check for quoted emails without removing the quotes